src/main/java/org/springframework/cloud/deployer/spi/kubernetes/LivenessCommandProbeCreator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.kubernetes;
+
+import org.springframework.cloud.deployer.spi.scheduler.ScheduleRequest;
+import org.springframework.cloud.deployer.spi.util.CommandLineTokenizer;
+import org.springframework.util.StringUtils;
+
+/**
+ * Creates a command based liveness probe
+ *
+ * @author Chris Schaefer
+ */
+class LivenessCommandProbeCreator extends CommandProbeCreator {
+	private final String propertyPrefix;
+
+	LivenessCommandProbeCreator(KubernetesDeployerProperties kubernetesDeployerProperties, ContainerConfiguration containerConfiguration) {
+		super(kubernetesDeployerProperties, containerConfiguration);
+		this.propertyPrefix = (containerConfiguration.getAppDeploymentRequest() instanceof ScheduleRequest) ?
+				"spring.cloud.scheduler.kubernetes.liveness" : "spring.cloud.deployer.kubernetes.liveness";
```

Comment:
actually to go a bit further we don't even need to do this instance of check as probes are only applied to streams so no need to select the proper prefix as we know it will be deployer. :1st_place_medal: for copy/paste ;-)